eng/Packages.Data.props

```diff
@@ -68,7 +68,7 @@
     <PackageReference Update="NUnit3TestAdapter" Version="3.13.0" />
     <PackageReference Update="Polly" Version="7.1.0" />
     <PackageReference Update="Portable.BouncyCastle" Version="1.8.5" />
-    <PackageReference Update="PublicApiGenerator" Version="9.3.0" />
+    <PackageReference Update="PublicApiGenerator" Version="10.0.1" />
```

Comment:
May I ask what this PublicApiGenerator package is and where it comes from?

Reply:
OK I see it comes from https://github.com/PublicApiGenerator/PublicApiGenerator/releases/tag/10.0.0, but I was not aware of us using this took for api approvals. @AlexGhiondea @pakrym is this a tool we are trying to use for tracking our APIs?

Reply:
No, we use ApiGen, this one is used in some track 1 sdks like ServiceBus.

Reply:
The package creates a csharp equivalent of the public API surface of the assembly it asserts. It prevents accidental breaking changes from sneaking in that would violate semver. I believe it was introduced by @sean.feldman early on.

Reply:
Out of curiosity what is ApiGen? 😁

Reply:
Ref assembly generation utility that dotnet/runtime uses https://github.com/dotnet/arcade/tree/master/src/Microsoft.DotNet.GenAPI

Reply:
That is cool. Honestly I mean if you would rather want to replace it with that feel free to just drop this PR. I sent this PR because I knew service bus is using it and I used this project to verify during various stages of beta whether we are on the right track. But quite frankly the tool is old and a set of hacks ok top of Code Dom...

Reply:
We are not using GenAPI for service bus yet so this PR seems fine.

Reply:
Except there are now some obsolete method usages that were not updated and cause build failure.

Reply:
Yeah because stupid me forgot to stage some changes. Like we say here in Switzerland the problem is usually between the chair and the keyboard ;)

Reply:
That's a universal issue, @danielmarbach 😂